depends/packages/qt.mk

```diff
@@ -123,33 +107,41 @@ $(package)_config_opts += -no-feature-undogroup
 $(package)_config_opts += -no-feature-undostack
 $(package)_config_opts += -no-feature-undoview
 $(package)_config_opts += -no-feature-vnc
-$(package)_config_opts += -no-feature-wizard
-$(package)_config_opts += -no-feature-xml
+# A workaround for https://bugreports.qt.io/browse/QTBUG-99957.
```

Comment:
This links to a bug report for Qt 6.2.2, where the last comment was 3 years ago. Should we leave a new comment that this is still an issue for all current Qt versions (I assume this is the case otherwise we'd have a patch to apply?).

Reply:
A new comment has been posted to that bug report.

Reply:
The issue can be observed in the following scenario on an `aarch64`  machine:
```
$ env HOSTS=x86_64-linux-gnu ./contrib/guix/guix-build
...
Building qt...
ninja: error: 'qtbase/src/platformsupport/input/CMakeFiles/InputSupportPrivate.dir/cmake_pch.hxx.gch', needed by 'qtbase/src/platformsupport/input/CMakeFiles/InputSupportPrivate.dir/InputSupportPrivate_autogen/mocs_compilation.cpp.o', missing and no known rule to make it
make: *** [funcs.mk:303: /bitcoin/depends/work/build/x86_64-linux-gnu/qt/6.7.3-4d1070613d2/./.stamp_built] Error 1
make: Leaving directory '/bitcoin/depends'
```

Reply:
It seems there is an issue with older CMake versions. I was able to reproduce it on Ubuntu `x86_64` while compiling natively:
```
$ cmake --version
cmake version 3.22.6

CMake suite maintained and supported by Kitware (kitware.com/cmake).
$ gmake -C depends -j 16 qt
<snip>
Building qt...
ninja: error: 'qtbase/src/platformsupport/input/CMakeFiles/InputSupportPrivate.dir/cmake_pch.hxx.gch', needed by 'qtbase/src/platformsupport/input/CMakeFiles/InputSupportPrivate.dir/InputSupportPrivate_autogen/mocs_compilation.cpp.o', missing and no known rule to make it
make: *** [funcs.mk:303: /home/hebasto/dev2/bitcoin/depends/work/build/x86_64-linux-gnu/qt/6.7.3-f92448b8cb4/./.stamp_built] Error 1
make: Leaving directory '/home/hebasto/dev2/bitcoin/depends'
```

Reply:
The problem occurs with CMake versions older than 3.25.0.

Reply:
Here is an upstream issue [fixed](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7759) in CMake 3.25.0 – https://gitlab.kitware.com/cmake/cmake/-/issues/23464

Reply:
Ok, so it looks like we should instead apply this patch: https://codereview.qt-project.org/c/qt/qtbase/+/633612 ?

Reply:
Thanks! The patch has been [applied](https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2745146334).

Reply:
It seems this patch is not enough...

Reply:
> It seems this patch is not enough...

[Fixed](https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2747921426) for macOS as well.